api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -117,7 +118,17 @@ public CohortDefinition getAnyEncounterDuringPeriodWithOccurrence() {
         cd.addParameter(new Parameter("encounterTypeList", "reporting.parameter.encounterTypeList", EncounterType.class, List.class, null));
         cd.addParameter(new Parameter("formList", "reporting.parameter.formList", Form.class, List.class, null));
         cd.addParameter(new Parameter("locationList", "reporting.parameter.location", Location.class, List.class, null));
-        return new MappedParametersCohortDefinition(cd, "onOrAfter", "startDate", "onOrBefore", "endDate", "encounterTypeList", "encounterTypes", "locationList", "locations", "atLeastCount", "atLeast", "atMostCount", "atMost", "formList", "forms");
+        cd.addParameter(new Parameter("timeQualifier", "reporting.parameter.timeQualifier", TimeQualifier.class));
+        return new MappedParametersCohortDefinition(cd,
+            "timeQualifier", "timeQualifier",
+            "onOrAfter", "startDate",
+            "onOrBefore", "endDate",
+            "encounterTypeList", "encounterTypes",
+            "locationList", "locations",
+            "atLeastCount", "atLeast",
+            "atMostCount", "atMost",
+            "formList", "forms"
+        );
     }
```

Comment:
@andela-aolaniran this looks good.  I'm not sure if you saw @djazayeri 's comment on another thread, but we had a discussion and we decided it would be best for the advanced libraries to simply expose the underlying parameter names as-is, so as to make it easier to determine from the java code what is the correct usage of the library.  So I think we can eliminate the use of the MappedParametersCohortDefinition here, and just return the EncounterCohortDefinition directly.